packages/cw-croncat-core/src/types.rs

```diff
@@ -562,6 +562,104 @@ impl Task {
     }
 }
 
+pub fn simulate_task(
+    task: TaskRequest,
+    funds: GenericBalance,
+    gas_base_fee: u64,
+    gas_action_fee: u64,
+    block_info: BlockInfo,
+    slot_granularity_time: u64,
+) -> Result<(u64, u64), CoreError> {
+    let mut occurrences: u64 = 0;
+    let interval = task.interval;
+    let mut gas_amount: u64 = gas_base_fee;
+
+    for action in task.actions.iter() {
+        gas_amount = gas_amount
+            .checked_add(action.gas_limit.unwrap_or(gas_action_fee))
+            .ok_or(CoreError::NoGasLimit {})?;
+    }
+    let boundary = BoundaryValidated::validate_boundary(task.boundary, &interval).unwrap();
+
+    match interval {
+        Interval::Once | Interval::Immediate => {
+            occurrences = occurrences.saturating_add(1);
+        }
+        Interval::Block(block) => {
+            let mut start_block: u64 = boundary.start.unwrap();
```

Comment:
```suggestion
            let mut start_block: u64 = boundary.start.unwrap_or(env.block.height);
```
I guess instead of panic we want some default value here?